test/integration/host_test.rb

```diff
@@ -533,6 +475,29 @@ def test_show_action_buttons
       host.reload
       assert_equal env1.name, host.environment.name
     end
+
+    test 'enables override of hostgroup defaults' do
+      original_environment = FactoryGirl.create(:environment)
+      original_hostgroup = FactoryGirl.create(:hostgroup,
+                                              :environment => original_environment)
+
+      # Clear host environment, then inherit it from hostgroup
+      @host.environment = nil
+      @host.hostgroup = original_hostgroup
+      @host.send(:assign_hostgroup_attributes, ['environment_id'])
```

Comment:
Or just assign the environment and host group, skip using this private method.  There's no extra information stored when inheriting, it only sets the environment association.